plugins/outputs/cloudrun/cloudrun.go

```diff
@@ -0,0 +1,197 @@
+  ## metrics which are set to the following so you don't have to.
+  ## defaultContentType   = "application/octet-stream"
+  ## defaultAccept        = "application/json"
+  ## defaultMethod        = http.MethodPost
+  ## The token is generated using the URL, json_file_location, and cloudrun_email you set in your conf file
+`
+
+const (
+	defaultClientTimeout = 5 * time.Second
+	defaultContentType   = "application/octet-stream"
+	defaultAccept        = "application/json"
+	defaultMethod        = http.MethodPost
+)
+
+type CloudRun struct {
+	URL          string            `toml:"url"`
+	Timeout      config.Duration   `toml:"timeout"`
+	Headers      map[string]string `toml:"headers"`
+	JSONSecret   string            `toml:"json_file_location"`
+	ConvertPaths bool              `toml:"convert_paths"`
+	Method       string
+	tls.ClientConfig
+
+	client      *http.Client
+	serializer  serializers.Serializer
+	accessToken string
+}
+
+func (cr *CloudRun) SetSerializer(serializer serializers.Serializer) {
+	cr.serializer = serializer
+}
+
+func (cr *CloudRun) createHTTPClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := cr.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: time.Duration(cr.Timeout),
+	}
+
+	return client, nil
+}
+
+func (cr *CloudRun) Connect() error {
+	if cr.Timeout == 0 {
+		cr.Timeout = config.Duration(defaultClientTimeout)
+	}
+
+	ctx := context.Background()
+	client, err := cr.createHTTPClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	cr.client = client
+
+	return nil
+}
+
+func (cr *CloudRun) Close() error {
+	return nil
+}
+
+func (cr *CloudRun) Description() string {
+	return "A plugin that is capable of transmitting metrics over HTTPS to a Cloud Run Wavefront proxy"
+}
+
+func (cr *CloudRun) SampleConfig() string {
+	return sampleConfig
+}
+
+func (h *CloudRun) Write(metrics []telegraf.Metric) error {
+	reqBody, err := h.serializer.SerializeBatch(metrics)
+	if err != nil {
+		return err
+	}
+
+	if err := h.write(reqBody); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cr *CloudRun) write(reqBody []byte) error {
+	var reqBodyBuffer io.Reader = bytes.NewBuffer(reqBody)
+	var err error
+	req, err := http.NewRequest(defaultMethod, cr.URL, reqBodyBuffer)
+	if err != nil {
+		return err
+	}
+
+	claims := jwt.StandardClaims{}
+	jwt.ParseWithClaims(cr.accessToken, &claims, func(token *jwt.Token) (interface{}, error) {
+		return nil, nil
+	})
+
+	if cr.accessToken == "" || !claims.VerifyExpiresAt(time.Now().Unix(), true) {
+		cr.accessToken, err = gcp.GetAccessToken(cr.JSONSecret, cr.URL)
+		if err != nil {
+			return err
+		}
+	}
+
+	bearerToken := fmt.Sprintf("Bearer %s", cr.accessToken)
+
+	req.Header.Set("User-Agent", internal.ProductToken())
+	req.Header.Set("Content-Type", defaultContentType)
+	req.Header.Set("Accept", defaultAccept)
```

Comment:
Please directly set the values here instead of using `default...` constants as this is the only user and you directly can see what is set instead of navigating the whole file.